src/graph/matrix/delta_matrix.rs

```diff
@@ -0,0 +1,510 @@
+                    x.delta_minus.set_always_hyper();
+                    Some(x)
+                } else {
+                    None
+                },
+                mutex: Some(CMutex::new()),
+            };
+            x.matrix.set_sparsity(GxB_SPARSE | GxB_HYPERSPARSE);
+            x.delta_plus.set_sparsity(GxB_HYPERSPARSE);
+            x.delta_plus.set_always_hyper();
+            x.delta_minus.set_sparsity(GxB_HYPERSPARSE);
+            x.delta_minus.set_always_hyper();
+            x
+        }
+    }
+
+    pub fn transposed(&mut self) -> Option<&mut Box<DeltaMatrix>> {
+        self.transposed.as_mut()
+    }
+
+    pub fn dirty(&self) -> bool {
+        self.dirty
+    }
+
+    fn set_dirty(
+        &mut self,
+        dirty: bool,
+    ) {
+        self.dirty = dirty;
+        if let Some(t) = self.transposed.as_mut() {
+            t.set_dirty(dirty);
+        }
+    }
+
+    pub fn m(&self) -> &SparseMatrix {
+        &self.matrix
+    }
+
+    pub fn dp(&self) -> &SparseMatrix {
+        &self.delta_plus
+    }
+
+    pub fn dm(&self) -> &SparseMatrix {
+        &self.delta_minus
+    }
+
+    pub fn nrows(&self) -> u64 {
+        self.matrix.nrows()
+    }
+
+    pub fn ncols(&self) -> u64 {
+        self.matrix.ncols()
+    }
+
+    pub fn nvals(&self) -> u64 {
+        self.matrix.nvals() + self.delta_plus.nvals() - self.delta_minus.nvals()
+    }
+
+    pub fn resize(
+        &mut self,
+        nrows_new: u64,
+        ncols_new: u64,
+    ) {
+        if let Some(t) = self.transposed.as_mut() {
+            t.resize(ncols_new, nrows_new);
+        }
+        self.matrix.resize(nrows_new, ncols_new);
+        self.delta_plus.resize(nrows_new, ncols_new);
+        self.delta_minus.resize(nrows_new, ncols_new);
+    }
+
+    pub fn remove_element_bool(
+        &mut self,
+        i: u64,
+        j: u64,
+    ) {
+        if let Some(t) = self.transposed.as_mut() {
+            t.remove_element_bool(j, i);
+        }
+
+        if let Some(true) = self.matrix.extract_element_bool(i, j) {
+            self.delta_minus.set_element_bool(i, j, true);
+        } else {
+            self.delta_plus.remove_element(i, j);
+        }
+        self.set_dirty(true);
+    }
+
+    pub fn set_element_bool(
+        &mut self,
+        i: u64,
+        j: u64,
+    ) {
+        if let Some(t) = self.transposed.as_mut() {
+            t.set_element_bool(j, i);
+        }
+
+        if let Some(true) = self.delta_minus.extract_element_bool(i, j) {
+            self.delta_minus.remove_element(i, j);
+            self.set_dirty(true);
+        } else if let None = self.matrix.extract_element_bool(i, j) {
+            self.delta_plus.set_element_bool(i, j, true);
+            self.set_dirty(true);
+        }
+    }
+
+    pub fn extract_element_bool(
+        &self,
+        i: u64,
+        j: u64,
+    ) -> Option<bool> {
+        if let Some(true) = self.delta_plus.extract_element_bool(i, j) {
```

Comment:
I think here unwrap_or_default() does the same effect, because bool's default is false, so if it is Some AND true you get a true, otherwise it's  a false